db-service/lib/cqn4sql.js

```diff
@@ -430,10 +430,11 @@ function cqn4sql(originalQuery, model = cds.context?.model || cds.model) {
 
       if (col.$refLinks.some(link => link.definition._target?.['@cds.persistence.skip'] === true)) return
 
+      const getName = col => col?.as || col?.ref?.at?.(-1)
```

Comment:
```suggestion
      const getName = col => col.as || col.ref?.at(-1)
```
According to former impl `col` was always defined.
And if `col.ref` exists, function `col.ref.at()` exists as well.